File: love2d/flexer2d/source/object.lua
local Object = {}
Object.__index = Object

function Object.new(x,y,width,height)
    local obj = setmetatable({}, Object)

    obj.x = x or 0
    obj.y = y or 0

    obj.origin = {
        x = 0,
        y = 0
    }

    obj.width = width or 1
    obj.height = height or 1

    obj.active = true
    obj.visible = true
    obj.debug = false

    obj.camera = nil

    obj.updateCallback = nil
    obj.drawCallback = nil
    obj.destroyCallback = nil

    return obj
end

function Object:setPosition(x,y)
    self.x, self.y = x, y
end
function Object:setHitbox(width,height)
    self.width, self.height = width, height
end
function Object:setOrigin(x,y)
    self.origin.x, self.origin.y = x, y
end

function Object:destroy()
    if self.destroyCallback then
        self:destroyCallback()
    end

    self.active = false
    self = nil
end
function Object:update(dt)
    if not self.active then
        return
    end

    if self.updateCallback then
        self:updateCallback(dt)
    end
end
function Object:draw()
    if not self.visible then
        return
    end

    if self.debug then
        love.graphics.rectangle('line', self.x, self.y, self.width, self.height)
        love.graphics.circle('fill', self.x + self.origin.x, self.y + self.origin.y, 1)
    end

    if self.drawCallback then
        self:drawCallback()
    end
end



return Object

File: pygame/flexer2d/modules/Object.py
class Object:

    '''
    TODO:
        - add debug hitbox viewer
    '''

    def __init__(self, x:int=0, y:int=0, width:int=1, height:int=1):
        self.set_position(x,y)
        self.set_size(width,height)
        self.set_scale(1, 1)

        self.active = True
        self.visible = True
        self.debug = False
                
        self.update = None
        self.draw = None

    def set_position(self, x:int, y:int):
        self.x = x
        self.y = y
    def add_position(self, x:int=0, y:int=0):
        self.x += x
        self.y += y

    def set_scale(self, x:int=1, y:int=1):
        self.scale_x = x
        self.scale_y = y
    def add_scale(self, x:int=0, y:int=0):
        self.scale_x += x
        self.scale_y += y

    def set_size(self, width:int=1, height:int=1):
        self.width = width
        self.height = height

    def __draw__(self):
        if not self.visible:
            return
        
        if self.draw != None:
            self.draw()
        
    def __update__(self):
        if not self.active:
            return
        
        if self.update != None:
            self.update()    